libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -67,6 +67,26 @@ assembly::Statement Parser::parseStatement()
 		return parseFunctionDefinition();
 	case Token::LBrace:
 		return parseBlock();
+	case Token::Switch:
+	{
+		assembly::Switch _switch = createWithLocation<assembly::Switch>();
+		m_scanner->next();
+		_switch.expression = make_shared<Statement>(parseExpression());
+		if (_switch.expression->type() == typeid(assembly::Instruction))
```

Comment:
Would be better to move this into `parseExpression` or create another function that calls `parseExpression` internally.